guis/panel/leak/panel_leak_utilities.py

```diff
@@ -92,3 +93,94 @@ def get_time(time_str):
         # df['PSI/degC'] = df["PRESSURE(PSI)"]/df["ROOM TEMPERATURE(C)"]
 
         return df
+
+
+################################################################################
+# READ INPUT
+# Read raw data from the database into a dataframe, structurally equivalent to that of ReadLeakRateFile, but with timestamps retained
+# TIME(DAYS)    FILLPSIA    RefPSIA PRESSURE(PSI)   BOX TEMPERATURE(C)  ROOM TEMPERATURE(C) Heater%
+################################################################################
+
+def readLeakDb(panel, tag):
+    leak_df = pd.DataFrame()
+    
+    #ensure that panel is type str
+    panel = str(panel)
+    
+    
+    con = sqlite3.connect('data/database.db')
+    cursor = con.cursor()
+
+    # acquire straw location
```

Comment:
There are several functions lying around the repository that get a procedure ID and/or straw location. Should use those.

I think you might have even worked with one or two of them here #147 or maybe #139?

If one doesn't exist in a common place, perhaps you could add it to `gui_utils.py`. I think that's a file. At worst, you can add it here.